src/main/java/wansbot/WansBot.java

```diff
@@ -0,0 +1,337 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import java.util.Scanner;
+
+import wansbot.storage.Storage;
+import wansbot.tasks.Deadlined;
+import wansbot.tasks.Events;
+import wansbot.tasks.InputEmptyException;
+import wansbot.tasks.NotANumMarkingException;
+import wansbot.tasks.TaskList;
+import wansbot.tasks.Todos;
+import wansbot.ui.UI;
+
+/**
+ * wansbot.Main class in the package which has all the logic of the Bot.
+ */
+public class WansBot {
+    private static TaskList userTaskList = new TaskList();
+    private static UI ui = new UI();
+    private static Storage storage = new Storage(ui);
+
+    /**
+     * Throws InputEmptyException when user doesn't input anything after command word.
+     */
+    protected static void emptyInput(String userInput) throws InputEmptyException {
+        if (userInput.strip().equalsIgnoreCase("todos")
+                || userInput.strip().equalsIgnoreCase("deadline")
+                || userInput.strip().equalsIgnoreCase("event")
+                || userInput.strip().equalsIgnoreCase("mark")
+                || userInput.strip().equalsIgnoreCase("unmark")
+                || userInput.strip().equalsIgnoreCase("remove")) {
+            throw new InputEmptyException(userInput);
+        }
+    }
+
+    /**
+     * Throws NumberFormatException and NotANumMarkingException when a non-number is inputted and when it's not
+     * a valid task number in the userTaskList.
+     */
+    private void checkNumInput(String userInput, int taskListSize) throws NumberFormatException,
+            NotANumMarkingException {
+        if (userInput.startsWith("unmark")) {
+            int posTask = Integer.parseInt(userInput.substring(7));
+            if (posTask > taskListSize || posTask < 1) {
+                throw new NotANumMarkingException(posTask);
+            }
+        } else if (userInput.startsWith("mark")) {
+            int posTask = Integer.parseInt(userInput.substring(5));
+            if (posTask > taskListSize || posTask < 1) {
+                throw new NotANumMarkingException(posTask);
+            }
+        } else if (userInput.startsWith("remove")) {
+            int posTask = Integer.parseInt(userInput.substring(7));
+            if (posTask > taskListSize || posTask < 1) {
+                throw new NotANumMarkingException(posTask);
+            }
+        }
+    }
+
+    /**
+     * Throws InputEmptyException when user does not input a deadline after "deadline /by".
+     */
+    private static void checkMissingInputDeadline(String userInput) {
+        String[] splitUser = userInput.split(" /by ", 2);
+        if (splitUser.length < 2) {
+            throw new InputEmptyException(userInput, "/by");
+        }
+    }
+
+    /**
+     * Throws custom InputEmptyException for events when either missing from date after /from, missing to date after
+     * /to or missing both.
+     */
+    protected static void missingInputEvent(String userInput) {
```

Comment:
Maybe we could use a verb to name this method similar to @wrjgold's comment (eg. checkMissingInputEvent)

Reply:
Yes sir, you're right.